framework/src/modules/chain/modules/blocks/chain.js

```diff
@@ -712,7 +696,7 @@ __private.popLastBlock = function(oldLastBlock, cb) {
 					oldLastBlock.transactions.reverse(),
```

Comment:
You are calling the undo method for transactions repeatedly. It should be only called once, while this mapSeries will repeat the undo call for all transactions in the block based on the number of transactions in the block.

Reply:
Yes I did the initial push with the basics working I'm about to commit the call to the undo only :)